Feat: per-agency card scheme support

thekaveman opened the pull request:
Closes #3140 

Closes #3000 since it was straightforward and relevant to these changes.

## Reviewing

### Model Admin

1. Login to the Admin as a superuser
2. Navigate to edit a `TransitAgency` instance
3. See the correctly ordered list of `supported_card_schemes`, with defaults
4. Confirm that at least one choice must be selected to Save
5. Confirm that any other combination of choices can be selected and Saved
<details>
<summary>Screenshot</summary>
<img width="874" height="225" alt="image" src="https://github.com/user-attachments/assets/9f4ab1ed-e14d-476b-8523-fc75f02d878e" />
</details>

### Help

1. Launch the app, visit the Help page directly without choosing a transit agency
2. See the list of all card schemes in the expected order
<details>
<summary>Screenshot</summary>
<img width="1019" height="776" alt="image" src="https://github.com/user-attachments/assets/81385e58-6cce-4322-a733-f84fd1283f28" />
</details>

3. Go back to the homepage and select a transit agency
4. Click the link at the bottom of the page to go to the Help page
5. See the list of the agency's supported card schemes
<details>
<summary>Screenshot</summary>
<img width="1013" height="677" alt="image" src="https://github.com/user-attachments/assets/d4a746ca-09a2-4146-a5f3-29b8f2c79923" />
</details>

### Eligibility start

1. Select an agency and any flow
2. See the list of the agency's supported card schemes
<details>
<summary>Screenshot</summary>
<img width="884" height="534" alt="image" src="https://github.com/user-attachments/assets/f3fc399a-654a-4f96-bfda-e27a384f7680" />
</details>

3. Click the `Learn more about contactless cards` modal link to open the modal
4. See the agency's list of supported card schemes
<details>
<summary>Screenshot</summary>
<img width="1033" height="808" alt="image" src="https://github.com/user-attachments/assets/14ce3c75-9255-4b2f-929f-f0bf18786ea7" />
</details>

### Enrollment index

#### Switchio

1. Configure Switchio QA enrollment
2. Pass eligibility for any flow
3. See the list of the agency's supported card schemes
<details>
<summary>Screenshot</summary>
<img width="905" height="689" alt="image" src="https://github.com/user-attachments/assets/a959a884-ad01-4962-8cc9-13baae290f90" />
</details>

#### Littlepay

1. Configure Littlepay QA enrollment
2. Pass eligibility for any flow
3. See the list of the agency's supported card schemes
<details>
<summary>Screenshot</summary>
<img width="886" height="701" alt="image" src="https://github.com/user-attachments/assets/fd932bbe-54c6-426e-b70b-fd77ac7d18b8" />
</details>

4. In the browser devtools, place a breakpoint in JavaScript on the call to the `littlepay` function
5. Reload the page to hit the breakpoint
6. Confirm the agency's supported card schemes are passed to `options.cardTypes`
<details>
<summary>Screenshot</summary>
<img width="1097" height="1150" alt="image" src="https://github.com/user-attachments/assets/df1171cb-725b-406e-9adb-684e89b5829a" />
</details>

#### CalFresh

1. Select and pass eligibility for a CalFresh flow with any transit processor
2. See the list of the agency's supported card schemes in the alert box
<details>
<summary>Screenshot</summary>
<img width="864" height="851" alt="image" src="https://github.com/user-attachments/assets/baa9b75e-d77c-40c2-9e94-5c897bda8f9e" />
</details>

### Spanish

Go through the sequences above, with Spanish as the selected language. Confirm the translations appear as expected with the agency's supported card scheme list.

<details>
<summary>Help - no agency</summary>
<img width="1049" height="594" alt="image" src="https://github.com/user-attachments/assets/722cdf07-6b55-4ecb-a6c9-e3d10128dd2f" />
</details>

<details>
<summary>Help - agency selected</summary>
<img width="1033" height="524" alt="image" src="https://github.com/user-attachments/assets/6ae3339b-dd9b-488a-883c-0d17e2412927" />
</details>

<details>
<summary>Eligibility start</summary>
<img width="873" height="944" alt="image" src="https://github.com/user-attachments/assets/b9b41b21-d20e-4c5b-b060-5f211ffa6995" />
</details>

<details>
<summary>Eligibility start - modal</summary>
<img width="1090" height="888" alt="image" src="https://github.com/user-attachments/assets/70b91d26-9e75-428e-b186-c353e78642ef" />
</details>

<details>
<summary>Enrollment index</summary>
<img width="872" height="685" alt="image" src="https://github.com/user-attachments/assets/145949cf-5b78-4118-8b45-0bc72be89436" />
</details>

### In-person

#### Switchio

_No changes_

#### Littlepay

1. Configure Littlepay QA enrollment
2. Log in to the Admin as an agency user
3. Select and pass any In-person flow
4. In the browser devtools, place a breakpoint in JavaScript on the call to the `littlepay` function
5. Reload the page to hit the breakpoint
6. Confirm the agency's supported card schemes are passed to `options.cardTypes`
<details>
<summary>Screenshot</summary>
<img width="894" height="1077" alt="image" src="https://github.com/user-attachments/assets/f5c05d35-632c-4c64-8d29-9b520d0ea79e" />
</details>

angela-tran commented:
:eyes: Reviewing this, just FYI!

angela-tran commented on tests/pytest/core/test_views.py:
```diff
@@ -183,12 +170,12 @@ def test_help_with_session_agency(client):
     assert response.status_code == 400
     assert "origin" in mocked_session_update.call_args.kwargs
-    assert mocked_session_update.call_args.kwargs["origin"] == mocked_active_agency.index_url
+    assert mocked_session_update.call_args.kwargs["origin"] == mocked_session_agency.return_value.index_url
```
I thought about this refactor and looked at the view / middleware code for a good bit. I agree with the conclusion that the fixture is redundant :+1:

thekaveman replied:
See the commit message on that for more of an explanation: c53e4d5be12ca4285bd46a44a440ea7d28a76f5a

thekaveman commented on benefits/core/context_processors.py:
```diff
@@ -124,3 +122,8 @@ def routes(request):
     return {"routes": app_routes.to_dict()}
+
+
+def unique_values(original_list):
+    # dict.fromkeys gets the unique values and preserves order
+    return list(dict.fromkeys(original_list))
```
Oh wow, didn't even notice... just moved it to the bottom to alphabetize. Good catch, will remove.

thekaveman replied:
Removed in the rebased [4847f10](https://github.com/cal-itp/benefits/pull/3158/commits/4847f108adf529198d6eb8d06e75244c6073b5d9)

thekaveman commented:
> This looks really nice!
> 
> Just one thing I noticed while testing in-person. We probably need to update [enrollment/index_littlepay](https://github.com/cal-itp/benefits/blob/main/benefits/in_person/templates/in_person/enrollment/index_littlepay.html#L64) (and line 74 in the same file too) since the template is still using `card_types` but the view's context is now passing `card_schemes`.

Ah, I had fixed that but forgot to push the commit! Thanks for catching it. Updated in [cde4631](https://github.com/cal-itp/benefits/pull/3158/commits/cde4631528fbea921bd8e8f63409d65e75bb98a4)
